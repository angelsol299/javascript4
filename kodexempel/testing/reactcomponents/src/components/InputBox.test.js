import { shallow } from 'enzyme';
import React from 'react';
import InputBox from './InputBox';

test('Rendering an input box', () => {
  const inputBox = shallow(<InputBox />);

  inputBox.find('input').simulate('change', { target: { value: 'Jonas' } });

  expect(inputBox.find('h1').text()).toContain('Hello');
  expect(inputBox.find('input')).toHaveLength(1);
  expect(inputBox.state('inputValue')).toMatch('Jonas');
});

test('Basic snapshot', () => {
  const inputBox = shallow(<InputBox />);

  expect(inputBox).toMatchSnapshot();
});