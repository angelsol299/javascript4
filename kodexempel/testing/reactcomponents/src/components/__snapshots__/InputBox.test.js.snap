// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic snapshot 1`] = `
<Fragment>
  <h1>
    Hello
  </h1>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</Fragment>
`;
